pkg/endpointslice/map.go

```diff
@@ -111,6 +117,20 @@ func (m *Map) Put(es *discovery.EndpointSlice) {
 		return
 	}
 
+	var localSlice *discovery.EndpointSlice
+	if cluster == m.localClusterID {
+		localSlices, err := localClient.EndpointSlices(es.Labels[constants.LabelSourceNamespace]).List(context.TODO(), metav1.ListOptions{
+			LabelSelector: labels.Set(map[string]string{
+				"kubernetes.io/service-name": es.Labels[constants.MCSLabelServiceName],
+			}).String(),
+		})
+		if err != nil || len(localSlices.Items) != 1 {
```

Comment:
Need to add a retry in case error is due to timeout. Those are not that uncommon if Kube API server is under heavy traffic.